File: packages/sdk-python/src/voltbots/enhance.py
"""Client-side prompt enhancement with Hivemind warnings."""

from __future__ import annotations

from .types import WarningsResponse


def enhance_prompt(prompt: str, warnings: WarningsResponse) -> str:
    """Prepend Hivemind warnings to a prompt.

    Uses the server-generated injection_text when available.
    No API call is made — this is a client-side helper.
    """
    if not warnings.injection_text:
        return prompt

    return f"{warnings.injection_text}\n\n---\n\n{prompt}"

File: packages/sdk-js/src/enhance.ts
import type { WarningsResponse } from './types.js';

/**
 * Prepend HiveMind warnings to a prompt.
 *
 * Uses the server-generated `injectionText` when available.
 * This is a client-side helper — no API call is made.
 */
export function enhancePrompt(
  prompt: string,
  warnings: WarningsResponse,
): string {
  if (!warnings.injectionText) return prompt;

  return `${warnings.injectionText}\n\n---\n\n${prompt}`;
}
